gammapy/spectrum/flux_point.py

```diff
@@ -80,17 +80,42 @@ def plot(self, ax=None, energy_unit='TeV',
 
         kwargs.setdefault('fmt', 'o')
         ax = plt.gca() if ax is None else ax
-        x = self['ENERGY'].quantity.to(energy_unit).value
-        y = self['DIFF_FLUX'].quantity.to(flux_unit).value
-        yh = self['DIFF_FLUX_ERR_HI'].quantity.to(flux_unit).value
-        yl = self['DIFF_FLUX_ERR_LO'].quantity.to(flux_unit).value
-        y, yh, yl = np.asarray([y, yh, yl]) * np.power(x, energy_power)
 
-        xh = self['ENERGY_ERR_HI'].quantity.to(energy_unit).value
-        xl = self['ENERGY_ERR_LO'].quantity.to(energy_unit).value
+        energy = self['ENERGY'].quantity.to(energy_unit)
+        energy_hi = self['ENERGY_ERR_HI'].quantity.to(energy_unit)
+        energy_lo = self['ENERGY_ERR_LO'].quantity.to(energy_unit)
+
+        flux = self['DIFF_FLUX'].quantity.to(flux_unit)
+        flux_lo = self['DIFF_FLUX_ERR_LO'].quantity.to(flux_unit)
+        flux_hi = self['DIFF_FLUX_ERR_HI'].quantity.to(flux_unit)
+
+        eunit = [_ for _ in flux.unit.bases if _.physical_type == 'energy'][0]
+        yunit = flux.unit * eunit ** energy_power
+        y = (flux * np.power(energy, energy_power)).to(yunit)
+        y_hi = (flux_hi * np.power(energy, energy_power)).to(yunit)
+        y_lo = (flux_lo * np.power(energy, energy_power)).to(yunit)
+
+        # plot flux points
+        is_ul = np.isnan(flux_lo).astype('bool')
```

Comment:
If you have time, could you expose this UL bool array as a property and document (in the class docstring?) this convention how ULs are handled by this class? Maybe even add an example showing how it works?

This will be important and used in other places as well, e.g. when doing chi^2 spectral model fits to flux points.